server/manager/src/main/java/org/apache/accumulo/manager/compaction/queue/CompactionJobPriorityQueue.java

```diff
@@ -113,6 +122,9 @@ public boolean equals(Object o) {
   private final AtomicLong dequeuedJobs;
   private final ArrayDeque<CompletableFuture<CompactionJobQueues.MetaJob>> futures;
   private long futuresAdded = 0;
+  private final Map<KeyExtent,Timer> jobAges;
+  private final Supplier<CompactionJobPriorityQueueStats> jobQueueStats;
+  private final AtomicReference<io.micrometer.core.instrument.Timer> jobQueueTimer;
```

Comment:
Could change this to wrap an optional and initially set it to empty(). That would void creating an optional object each time this is used.

```suggestion
  private final AtomicReference<Optional<io.micrometer.core.instrument.Timer>> jobQueueTimer;
```